kivy/base.py

```diff
@@ -387,21 +387,26 @@ def idle(self):
         Clock.tick()
 
         # read and dispatch input from providers
-        self.dispatch_input()
+        if not self.quit:
```

Comment:
Can we do only one `if not self.quit:`?

Reply:
Probably not, because each of these calls (`self.dispatch_input()`, `Builder.sync()`,...) can cause `app.stop()` which results in `App.get_current_app()` returning `None` which raises an error. 

So a comment about `if not self.quit:` would be good to add.